src/full/Agda/Syntax/Translation/ConcreteToAbstract.hs

```diff
@@ -677,7 +668,11 @@ instance ToAbstract ResolveQName where
   type AbsOfCon ResolveQName = ResolvedName
   toAbstract (ResolveQName x) = resolveName x >>= \case
     UnknownName -> notInScopeError x
-    q -> return q
+    q -> do
+      case q of
+        DefinedName _ d _ -> addGeneralizable d
```

Comment:
Mmh, I am not sure I like this side effect here.
What does it have to do with `ResolveQName`?
It seems a mere accident that this is the right place for adding a generalized variable here.
Future code changes might invalidate this _chance correctness_.

If this is the right place after all, there should be a verbal justification in form of a comment that explains why generalizable are added whenever a `ResolveQName` is further processed by the scope checker.

Reply:
`ResolveQName` is currently only used to scope check builtin pragmas, and it seems to me that it should behave the same way as `OldQName`, which is used when scope checking other pragmas.

Generalizable `OldQName`s are collected for later generalization whenever they're resolved, and an error is raised if they're in a place where generalization isn't supported. Since pragmas don't support generalization, generalized names always raise errors in pragmas. If `ResolveQName` were to be used in a place where generalization is supported, I feel it should also be collected for later generalization like `OldQName`.

(It might even be desirable to add `raiseWarningsOnUsage` for `ResolveQName` like for `OldQName`, so that deprecated names also raise warnings in builtin pragmas.)

Maybe `ResolveQName` could be renamed to `ResolveOldQName` to make the connection clear? Or the instance could just be turned into a local helper function.

Reply:
> Or the instance could just be turned into a local helper function.

I turned it into a helper function.

Reply:
Looks good to me.